avro-fastserde/src/main/java/com/linkedin/avro/fastserde/PrimitiveFloatList.java

```diff
@@ -212,14 +258,28 @@ public Float remove(int i) {
     if (i >= size) {
       throw new IndexOutOfBoundsException("Index " + i + " out of bounds.");
     }
+    cacheFromByteBuffer();
     Float result = elements[i];
     --size;
     System.arraycopy(elements, i + 1, elements, i, (size - i));
     elements[size] = 0;
     return result;
   }
 
+  private void cacheFromByteBuffer() {
+    if (!isCached) {
```

Comment:
Looks like this function is not thread-safe, right? Can we add a synchronized block inside this if, which re-checks the `if (!isCached)` condition once inside the synchronized?